torch/lib/c10d/ProcessGroupMPI.cpp

```diff
@@ -144,12 +147,26 @@ int ProcessGroupMPI::AsyncWork::sourceRank() const {
 
 bool ProcessGroupMPI::AsyncWork::wait(std::chrono::milliseconds /* unused */) {
```

Comment:
does the user has to call wait()? What if for some reason the use does something like this:

while(!work.isComplete()) {  do_more_io_work(); }

Reply:
I guess this is technically possible, but if we are designing this similar to RPC API, then we only guarantee event is profiled if user issues a call to `work.wait()` since only after that are we guaranteed that the work has been completed. 

Also, the manual calling of this in `AsyncWork::wait()` is only for send/recv profiling, all other MPI collectives go through the same path as gloo (i.e. call `ProcessGroup::Work::finish`). I guess if user does the following:
```
work = mpi_send(...)
# do a bunch of long-runningother work
work.wait()
```
then the profiling time may be inaccurate, since it would be an overestimate and include the time for the other work in between.

Reply:
If that's the case, do we have documented this limitation somewhere? I mean the fact that wait is needed for send/recv inorder for profile to work.

Reply:
Created an issue mentioning to document this: https://github.com/pytorch/pytorch/issues/53684. We currently don't have much documentation around profiling distributed collectives since the work is landing, but after the main diffs are in, we will add documentation before releasing this feature.